core/store/src/trie/trie_storage.rs

```diff
@@ -548,12 +562,23 @@ impl TrieStorage for TrieCachingStorage {
     }
 }
 
+/// This method reads the shard UID mapping from db.
+/// See `MappedShardUID` documentation for explanation.
+pub fn map_shard_uid(store: &Store, shard_uid: ShardUId) -> MappedShardUId {
+    match store.get_ser::<ShardUId>(DBCol::ShardUIdMapping, &shard_uid.to_bytes()) {
+        Ok(Some(mapped_shard_uid)) => mapped_shard_uid,
+        // TODO(reshardingV3) Think about how to handle it the best way
+        _ => shard_uid,
```

Comment:
+1 best to think about what are the invariants and handle it accordingly. 

If the invariant is that the mapping should always be populated for all shards then returning an error seems quite reasonable.